pepsico/maproom_utilities.py

```diff
@@ -0,0 +1,413 @@
+        than `bbox` or a contained admin level (1) with 2 geometries partitioning
+        level 0
+    
+    Returns
+    -------
+    df : pandas.DataFrame
+        a pd.DF with columns "label" (dtype=string),
+        "key" (string or int depending on the table),
+        and "the_geom" (shapely.Geometry)
+
+    See Also
+    --------
+    shapely.geometry.Polygon
+
+    Notes
+    -----
+    A level 0 contained into bbox is necessary to test the clipping feature since
+    `bbox` is also used to generate the fake data.
+    """
+    west, south, east, north = bbox
+    assert (south + 0.25) <= (north - 0.5), (
+        "Please extend latitudinal domain of bbox"
+    )
+    if east < west :
+        assert (west + 0.25) >= (east - 0.5), (
+            "Please extend longitudinal domain of bbox"
+        )
+    else :
+        assert (west + 0.25) <= (east - 0.5), (
+            "Please extend longitudinal domain of bbox"
+        )
+    west = west + 0.25
+    south = south + 0.25
+    east = east - 0.5
+    norht = north - 0.5
+    if level == 0 :
+        df = pd.DataFrame({"label" : ["Guyane"], "key": [0], "the_geom": [Polygon([
+            [west, south], [west, north], [east, north], [east, south]
+        ])]})
+    elif level == 1 : #2 triangles partitioning level-0 box at its SW-NE diagnonal
+        df = pd.DataFrame({"label" : ["NW", "SE"], "key": [1, 2],"the_geom": [
+            Polygon([[west, south], [west, north], [east, north]]),
+            Polygon([[west, south], [east, north], [east, south]]),
+        ]})
+    else:
+        raise Exception("level must be 0 or 1")
+    return df
+
+
+def make_adm_overlay(
+    adm_name, adm_id, adm_geojson, adm_color, adm_lev, adm_weight, is_checked=False
+):
+    """ Draw a dlf.Overlay of a dlf.GeoJSON for Maprooms admin
+
+    Parameters
+    ----------
+    adm_name: str
+        name to give the dlf.Overlay
+    adm_id: str
+        unique id of dlf.Overlay
+    adm_geojson: dict
+        GeoJSON of the admin
+    adm_color: str
+        color to give the dlf.GeoJSON
+    adm_lev: int
+        index to give the dlf.GeoJSON's id
+    adm_weight: int
+        weight to give the dlf.GeoJSON
+    is_checked: boolean, optional
+        whether dlf.Overlay is checked or not (default: not)
+
+    Returns
+    -------
+    adm_layer : dlf.Overlay
+        dlf.GeoJSON overlay with the given parameters
+
+    See Also
+    --------
+    calc.sql2GeoJSON for the format of `adm_geojson`,
+    dash_leaflet.Overlay, dash_leaflet.GeoJSON
+    """
+    border_id = {"type": "borders_adm", "index": adm_lev}
+    return dlf.Overlay(
+        dlf.GeoJSON(
+            id=border_id,
+            data=adm_geojson,
+            options={
+                "fill": True,
+                "color": adm_color,
+                "weight": adm_weight,
+                "fillOpacity": 0,
+            },
+        ),
+        name=adm_name,
+        id=adm_id,
+        checked=is_checked,
+    )
```

Comment:
down to here is copied from enacts where we split the geometry-wise functionality. With the addition of the multiple admin sets in pepsico case.